### ISSUE TITLE
App crashing on call auto accept in release mode apk, but working fine in debugging

### ISSUE THREAD

**Issue by umairm1alik** [twilio/twilio-voice-react-native#384]
I am implementing auto accept call on call receive the moment it accepted automatically my app crashes but working fine if I manually accept the call, i am receiving this error in the logcat

 E  FATAL EXCEPTION: mqt_native_modules Process: com.intelytics, PID: 8619 java.lang.NullPointerException at java.util.Objects.requireNonNull(Objects.java:207) at com.twiliovoicereactnative.CallListenerProxy.onConnected(CallListenerProxy.java:99) at com.twilio.voice.Call$1.lambda$onConnected$1$com-twilio-voice-Call$1(Call.java:680) at com.twilio.voice.Call$1$$ExternalSyntheticLambda0.run(Unknown Source:4) at android.os.Handler.handleCallback(Handler.java:938) at android.os.Handler.dispatchMessage(Handler.java:99) at android.os.Looper.loopOnce(Looper.java:201) at android.os.Looper.loop(Looper.java:288) at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:233) at java.lang.Thread.run(Thread.java:1012)

**Comment by mhuynh5757** [comment]
Hi @umairm1alik, thanks for bringing this to our attention. Please fill out the whole issue template, the version details help us debug these problems. What versions are you using, are you using Expo, etc.

**Comment by mhuynh5757** [comment]
Hi @umairm1alik I'd like to close this issue as stale for now, please feel free to add the additional info if you're still experiencing this issue and reopening the issue.

**Comment by umairm1alik** [comment]
@mhuynh5757 sorry for the late response , I am using "@twilio/voice-react-native-sdk": "^1.0.0", and  "react-native": "0.74.1" cli.
on call receive when I am accepting calls automatically my app crashes with the above error on release while it is working fine in debug mode.
Do I have to add extra permission? I am accepting calls without user consent.

`newVoice.on('callInvite', async (callInvite: any) => {
       
        // console.log('Incoming call invite:', callInvite);
        // setTimeout(() => {
        //   try {
        //     callInvite.accept();
        //   } catch (error) {
        //     console.error('Error accepting call:', error);
        //   }
        // }, 3000); // 1000 ms = 1 second
      });`

**Comment by mhuynh5757** [comment]
Hi @umairm1alik could you please let me know what version of Android you're targeting, and the API Version that you're building for? Also, you could try to update to the latest version of the RN SDK `1.1.0` and see if this resolves your issue.

**Comment by umairm1alik** [comment]
@mhuynh5757 I targeting android versions greater than 11, currently on which I am getting issue is android 12.
sure I will let you know after updating to 1.1.0

**Comment by umairm1alik** [comment]
@mhuynh5757 I have updated to sdk 1.1.0 , at the first time I initiated a call and auto-accept executed my app crashed with the below logcat, but working fine when I again opened the app and triggered the call.

`024-08-26 16:20:56.142 19467-19497 EffectHAL               android.hardware.audio.service       W  Effect 0xe79e8690 command SET_CONFIG returned status: Invalid argument
2024-08-26 16:20:56.143 19467-19497 EffectHAL               android.hardware.audio.service       W  Effect 0xe79e7870 command SET_CONFIG returned status: Invalid argument
2024-08-26 16:20:56.146 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioEffectsExternal: enable(audioSession=1238129)
2024-08-26 16:20:56.150 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioEffectsExternal: AcousticEchoCanceler: was enabled, enable: true, is now: enabled
2024-08-26 16:20:56.153 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioEffectsExternal: NoiseSuppressor: was enabled, enable: true, is now: enabled
2024-08-26 16:20:56.153 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioRecordExternal: AudioRecord: session ID: 1238129, channels: 1, sample rate: 48000
2024-08-26 16:20:56.153 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioRecordExternal: AudioRecord: buffer size in frames: 3840
2024-08-26 16:20:56.154 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioRecordExternal: Number of active recording sessions: 0
2024-08-26 16:20:56.154 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioRecordExternal: startRecording
2024-08-26 16:20:56.163 19467-19497 audio_hw_primary        android.hardware.audio.service       D  adev_create_audio_patch: enter: num sources 1, num_sinks 1, handle 0
2024-08-26 16:20:56.163 19467-19497 audio_hw_primary        android.hardware.audio.service       D  adev_create_audio_patch: Added a new patch with handle 2179
2024-08-26 16:20:56.165  1861-4842  SoundTrigg...areLogging system_server                        I  setCaptureState[this=com.android.server.soundtrigger_middleware.SoundTriggerMiddlewareImpl@f132df3, client=null](true)
2024-08-26 16:20:56.167 19467-19497 audio_hw_primary        android.hardware.audio.service       D  AEC enable 1
2024-08-26 16:20:56.167 19467-19497 audio_hw_primary        android.hardware.audio.service       D  select_devices for use case (audio-playback-voip)
2024-08-26 16:20:56.167 19467-19497 msm8974_platform        android.hardware.audio.service       D  is_operator_tmus: tmus mccmnc 41001
2024-08-26 16:20:56.167 19467-19497 audio_hw_primary        android.hardware.audio.service       D  select_devices: changing use case audio-playback-voip output device from(1: handset, acdb 7) to (25: voice-handset, acdb 7)
2024-08-26 16:20:56.167 19467-19497 audio_hw_primary        android.hardware.audio.service       D  disable_audio_route: reset and update mixer path: audio-playback-voip
2024-08-26 16:20:56.168 19774-20471 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioRecordExternal: scheduleLogRecordingConfigurationsTask
2024-08-26 16:20:56.168 19774-20570 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioRecordExternal: AudioRecordThread@[name=AudioRecordJavaThread, id=3824]
--------- beginning of crash
2024-08-26 16:20:56.168 19774-20570 tvo.webrtc.Logging      com.intelytics                       I  WebRtcAudioRecordExternal: doAudioRecordStateCallback: START
2024-08-26 16:20:56.170 19774-19839 AndroidRuntime          com.intelytics                       E  FATAL EXCEPTION: mqt_native_modules
                                                                                                    Process: com.intelytics, PID: 19774
                                                                                                    java.lang.NullPointerException
                                                                                                    	at java.util.Objects.requireNonNull(Objects.java:207)
                                                                                                    	at com.twiliovoicereactnative.CallListenerProxy.onConnected(CallListenerProxy.java:99)
                                                                                                    	at com.twilio.voice.Call$1.lambda$onConnected$1$com-twilio-voice-Call$1(Call.java:661)
                                                                                                    	at com.twilio.voice.Call$1$$ExternalSyntheticLambda0.run(Unknown Source:4)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:233)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
2024-08-26 16:20:56.171 19774-19774 Twilio:cla...lang.Class com.intelytics                       I  [Platform] {"timestamp_ms":1724671256170,"client_name":"abcd","temp_call_sid":"abcd","call_sid":"xyz","message_sid":"xyz","sdk_version":"1.1.0","platform":"android","direction":"INCOMING","selected_region":"gll","gateway":"jkhngk","region":"ASIAPAC_SINGAPORE","logger_type":"default"}
2024-08-26 16:20:56.171  1181-2566  sensors-hal             and....sensors@2.0-service.multihal  I  [ssc_utils]get_qmi_debug_flag:243, support_qmi_debug : false
2024-08-26 16:20:56.174 19774-19774 Twilio:cla...lang.Class com.intelytics                       I  [Platform] {"timestamp_ms":1724671256173,"client_name":"xyz","temp_call_sid":"abcd","call_sid":"xyz","message_sid":"xyz","sdk_version":"1.1.0","platform":"android","direction":"INCOMING","selected_region":"gll","gateway":"xuz","region":"ASIAPAC_SINGAPORE","logger_type":"default"}
2024-08-26 16:20:56.175 19774-19774 Twilio:cla...lang.Class com.intelytics                       I  [Platform] {"timestamp_ms":1724671256174,"client_name":"xyz","temp_call_sid":"abcfs","call_sid":"xyz","message_sid":"xyz","sdk_version":"1.1.0","platform":"android","direction":"INCOMING","selected_region":"gll","gateway":"dwgwg","region":"ASIAPAC_SINGAPORE","logger_type":"default"}
2024-08-26 16:20:56.180  1861-4842  ActivityTaskManager     system_server                        W    Force finishing activity com.intelytics/.MainActivity
2024-08-26 16:20:56.180  1861-20578 DropBoxManagerService   system_server                        I  add tag=data_app_crash isTagEnabled=true flags=0x2
2024-08-26 16:20:56.184 19467-19497 soundtrigger            android.hardware.audio.service       D  audio_extn_sound_trigger_update_stream_status: uc_info->id 33 of type 0 for Event 2, with Raise=0
2024-08-26 16:20:56.184 19467-19497 hardware_info           android.hardware.audio.service       D  hw_info_append_hw_type : device_name = handset
2024-08-26 16:20:56.184 19467-19497 audio_hw_primary        android.hardware.audio.service       D  disable_snd_device: snd_device(1: handset)
2024-08-26 16:20:56.194 19467-19497 soundtrigger            android.hardware.audio.service       I  audio_extn_sound_trigger_update_device_status: device 0x1 of type 0 for Event 0, with Raise=0
2024-08-26 16:20:56.194 19467-19497 msm8974_platform        android.hardware.audio.service       I  platform_check_and_set_codec_backend_cfg:becf: afe: bitwidth 16, samplerate 48000 channels 2, backend_idx 0 usecase = 33 device (voice-handset)
2024-08-26 16:20:56.194 19467-19497 msm8974_platform        android.hardware.audio.service       I  platform_check_and_set_codec_backend_cfg: new_snd_devices[0] is 25
2024-08-26 16:20:56.194 19467-19497 msm8974_platform        android.hardware.audio.service       I  platform_check_codec_backend_cfg:becf: afe: bitwidth 16, samplerate 48000 channels 2, backend_idx 0 usecase = 33 device (voice-handset)
2024-08-26 16:20:56.195 19467-19497 msm8974_platform        android.hardware.audio.service       D  platform_check_codec_backend_cfg:becf: updated afe: bitwidth 16, samplerate 48000 channels 2,backend_idx 0 usecase = 33 device (voice-handset)
2024-08-26 16:20:56.195 19467-19497 msm8974_platform        android.hardware.audio.service       I  platform_check_codec_backend_cfg:becf: afe: Codec selected backend: 0 updated bit width: 16 and sample rate: 48000
2024-08-26 16:20:56.195 19467-19497 audio_hw_primary        android.hardware.audio.service       D  check_usecases_codec_backend:becf: force routing 0
2024-08-26 16:20:56.195 19467-19497 audio_hw_primary        android.hardware.audio.service       D  check_usecases_codec_backend:becf: (78) check_usecases curr device: voice-handset, usecase device:handset backends match 1
2024-08-26 16:20:56.195 19467-19497 audio_hw_primary        android.hardware.audio.service       D  check_usecases_codec_backend:becf: check_usecases num.of Usecases to switch 0
2024-08-26 16:20:56.195 19467-19497 hardware_info           android.hardware.audio.service       D  hw_info_append_hw_type : device_name = voice-handset
2024-08-26 16:20:56.195 19467-19497 audio_hw_primary        android.hardware.audio.service       D  enable_snd_device: snd_device(25: voice-handset)
2024-08-26 16:20:56.195 19467-19497 msm8974_platform        android.hardware.audio.service       D  platform_get_island_cfg_on_device:island cfg status on snd_device = (voice-handset 0)
2024-08-26 16:20:56.196 19467-19497 soundtrigger            android.hardware.audio.service       I  audio_extn_sound_trigger_update_device_status: device 0x19 of type 0 for Event 1, with Raise=0
2024-08-26 16:20:56.196 19467-19497 audio_route             android.hardware.audio.service       D  Apply path: voice-handset
2024-08-26 16:20:56.196  1861-2239  BroadcastQueue          system_server                        W  Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2024-08-26 16:20:56.197 19774-19839 Process                 com.intelytics                       I  Sending signal. PID: 19774 SIG: 9
2024-08-26 16:20:56.208 19467-19497 soundtrigger            android.hardware.audio.service       D  audio_extn_sound_trigger_update_stream_status: uc_info->id 33 of type 0 for Event 3, with Raise=0
2024-08-26 16:20:56.208 19467-19497 audio_hw_utils          android.hardware.audio.service       D  audio_extn_utils_send_app_type_cfg: usecase->out_snd_device voice-handset
2024-08-26 16:20:56.208 19467-19497 audio_hw_utils          android.hardware.audio.service       I  send_app_type_cfg_for_device PLAYBACK app_type 69946, acdb_dev_id 7, sample_rate 48000, snd_device_be_idx 119
2024-08-26 16:20:56.208 19467-19497 audio_hw_waves          android.hardware.audio.service       D  ma_set_device: 33 is not supported usecase
2024-08-26 16:20:56.208 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_audio_cal, acdb_id = 7, path = 0, app id = 0x1113a, sample rate = 48000, afe_sample_rate = 48000
2024-08-26 16:20:56.208 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_asm_topology
2024-08-26 16:20:56.208 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_STREAM_TOPOLOGY_ID
2024-08-26 16:20:56.208 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_adm_topology
2024-08-26 16:20:56.208 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_COMMON_TOPOLOGY_ID
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_audtable
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_INSTANCE_COMMON_TABLE_SIZE
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_INSTANCE_COMMON_TABLE
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> AUDIO_SET_AUDPROC_CAL cal_type[11] acdb_id[7] app_type[69946]
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_audvoltable
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_INSTANCE_GAIN_DEP_STEP_TABLE_SIZE
2024-08-26 16:20:56.209 19467-19497 android.ha...io.service android.hardware.audio.service       D  Failed to fetch the lookup information of the device 00000007 
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  Error: ACDB_CMD_GET_AUDPROC_INSTANCE_GAIN_DEP_STEP_TABLE_SIZE Returned = -19
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_INSTANCE_GAIN_DEP_STEP_TABLE, vol index 0
2024-08-26 16:20:56.209 19467-19497 android.ha...io.service android.hardware.audio.service       D  Failed to fetch the lookup information of the device 00000007 
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  Error: ACDB AudProc vol returned = -19
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> AUDIO_SET_VOL_CAL cal type = 12
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_INSTANCE_STREAM_TABLE_SIZE
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_audstrmtable
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AUDPROC_INSTANCE_STREAM_TABLE
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> audstrm_cal->cal_type.cal_data.cal_size = 20
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_afe_topology
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AFE_TOPOLOGY_ID
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> GET_AFE_TOPOLOGY_ID for adcd_id 7, Topology Id 112fc
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_afe_cal
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AFE_INSTANCE_COMMON_TABLE_SIZE
2024-08-26 16:20:56.209 19467-19497 android.ha...io.service android.hardware.audio.service       D  Failed to fetch the lookup information of the device 00000007 
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  Error: ACDB_CMD_GET_AFE_INSTANCE_COMMON_TABLE_SIZE Returned = -19
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_CMD_GET_AFE_INSTANCE_COMMON_TABLE
2024-08-26 16:20:56.209 19467-19497 android.ha...io.service android.hardware.audio.service       D  Failed to fetch the lookup information of the device 00000007 
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  Error: ACDB AFE returned = -19
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> AUDIO_SET_AFE_CAL cal_type[16] acdb_id[7]
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> send_hw_delay : acdb_id = 7 path = 0
2024-08-26 16:20:56.209 19467-19497 ACDB-LOADER             android.hardware.audio.service       D  ACDB -> ACDB_AVSYNC_INFO: ACDB_CMD_GET_DEVICE_PROPERTY
2024-08-26 16:20:56.209 19467-19497 audio_hw_primary        android.hardware.audio.service       D  enable_audio_route: apply mixer and update path: audio-playback-voip handset
2024-08-26 16:20:56.209 19467-19497 audio_route             android.hardware.audio.service       D  Apply path: audio-playback-voip handset
2024-08-26 16:20:56.230  1181-1332  CHRE                    and....sensors@2.0-service.multihal  D  @ 2433775.558: [ActivityPlatform] type 4, confidence 95
2024-08-26 16:20:56.237 19467-19471 audio_hw_primary        android.hardware.audio.service       D  NS enable 1
2024-08-26 16:20:56.241 19467-20582 audio_hw_primary        android.hardware.audio.service       D  start_input_stream: enter: stream(0xecb070f0)usecase(34: audio-record-voip)
2024-08-26 16:20:56.241 19467-20582 audio_hw_primary        android.hardware.audio.service       D  select_devices for use case (audio-record-voip)`

**Comment by mhuynh5757** [comment]
Hey @umairm1alik could you please reformat your crash log into something more readable. It's hard to parse as-is.

**Comment by mhuynh5757** [comment]
Hi all, going to close this issue as stale for now. Please feel free to reopen the issue if you are still in need of help.
